### ISSUE TITLE
Canadian Cropland dataset experiments

### ISSUE THREAD

**Issue by gabrieltseng** [antofuller/CROMA#4]
Hi,

Thanks for this work, and for the thorough evaluations! Will the code used to evaluate CROMA on the downstream datasets be shared? In addition, will the (modified) data itself be shared? 

I also have some implementation questions:

1. Regarding the Canadian Cropland Results:
    1. For the Canadian Cropland results in Table 1 of the paper, was the RGB data used (as is shared in the [Google drive](https://drive.google.com/drive/folders/1mNI8B5EMk0Xgvx2Pc9ztnQRaW9pXh8yb)), or were all S2 bands used from the raw data?

    2. [This line](https://github.com/antofuller/CROMA/blob/2fd83cdd7a60bd32a9b9b53bad43624f8bf31421/main_croma.py#L14) implies CROMA is designed to ingest 120x120 images; were the 64x64 images from the cropland dataset resized to 120x120?

    3. The paper describes combining the validation and test sets into a single evaluation set; in this case, how were hyperparameters for finetuning selected?

2. Regaring the MARIDA results:
    1. How were the 256x256 images cropped into 96x96 images? (since 96 does not divide cleanly into 256)

**Comment by antofuller** [comment]
Thanks for your interest and thorough reading of CROMA. 

By "code used to evaluate CROMA," do you mean the entire pipeline of loading in data, preprocessing, and finetuning/probing/kNN/K-Means? I point to implementations in the appendix for kNN and K-Means. But I'll try to upload the rest of the pipeline as well. If you need something quick, please email me at anthonyfuller@cmail.carleton.ca

Yes, I will share the preprocessed datasets soon. Again, let me know if you need something quick, for example, if you are preparing a paper that requires it. 

Canadian Cropland:

1. I used S2 bands from the raw data in that Google Drive—all experiments in the paper used S2 and/or S1 data.
2. I always resized images to match those on which the model was pretrained. (The only exception is when evaluating SatMAE on BigEarthNet, I found better results than those listed in their original paper by resizing position embeddings.) This meant resizing Canadian Cropland images up to 120x120. 
3. Good question. I followed prior work by only reporting results on validation sets in the paper. Hyperparameters were chosen by sweeping through base learning rates  {3e-5, 5e-5, 8e-5, 1e-4, 3e-4, 5e-4, 8e-4, 1e-3} for finetuning and {1, 2, 3, 4, 5, 6, 7, 8, 9}e{-4, -3, -2} for nonlinear and linear probing. To ensure fair comparisons, I performed this sweep for every model that is listed in a table in CROMA. I felt the need to undertake this extensive evaluation because prior models were not evaluated under identical conditions (or even data splits!). This being said, I **strongly** support a move to more consistent evaluations in DL for RS that include consistent validation and test sets.

MARIDA:

1. I will have to get back to you.
2. SatMAE uses 8x8px patches for Sentinel-2 data (see SatMAE's appendix A.3.3). I trained linear probes to output a class prediction for each pixel—so a linear map from model_dim -> pixels_per_patch. (Since SatMAE splits spectral bands into multiple co-located patches, I concatenated patches that arose from the same location before the linear probe.) In reality, the models  typically output the same class predictions for all pixels in the patch. So, I very likely would have gotten the same rank by converting this into a patch classification problem.
3. Same as 3 from above.

Thanks again for your interest in this work and let me know if you have further questions :)

**Comment by gabrieltseng** [comment]
Thank you for your quick reply. Just to clarify, when you say "validation set" you are referring to the combined validation and test set (which is described as an "evaluation set" in the paper)?

**Comment by antofuller** [comment]
Yes, that is correct.
